app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperDownloader.kt

```diff
@@ -65,6 +65,36 @@ class WallpaperDownloader(
         }
     }
 
+    /**
+     * Attempts to download the thumbnail for a wallpaper.
+     *
+     * @param wallpaper The wallpaper for which we want to download the thumbnail.
+     * @param onFailure onFailure callback invoked when the thumbnail fails to download.
+     */
+    fun downloadThumbnail(wallpaper: Wallpaper, onFailure: (Throwable) -> Unit) {
```

Comment:
will this conflict with https://github.com/mozilla-mobile/fenix/pull/26562?

Reply:
Yes, I'll rebase and remove this function once the work for #26562 is merged. Since we'll be waiting for the PR to land I'll add the do-not-land label here.